go/vt/vtgate/planbuilder/physical/subquery_planning.go

```diff
@@ -141,12 +194,51 @@ func tryMergeSubQueryOp(
 	var err error
 	switch outerOp := outer.(type) {
 	case *Route:
-		if shouldTryMergingSubquery(outerOp, subq) {
-			merged, err = tryMerge(ctx, outerOp, subq, joinPredicates, merger)
+		subqueryRoute, isRoute := subq.(*Route)
+		if !isRoute {
+			return nil, nil
+		}
+
```

Comment:
the switch case has become too long, may be better to have some small functions.